README.md

```diff
@@ -34,139 +34,275 @@ For the moment, only one client can be instantiated globally per process.
+### Response and Errors
+
+The response is an empty array if all transfers were created
+successfully. If the response is non-empty, each object in the
+response array contains error information for an transfer that
+failed. The error object contains an error code and the index of the
+transfer in the request batch.
+
 ```js
-  enum TransferFlags {
-    linked = (1 << 0),
-    pending = (1 << 1),
-    post_pending_transfer = (1 << 2),
-    void_pending_transfer = (1 << 3)
+const errors = await client.createTransfers([transfer1, transfer2, transfer3]);
+
+// errors = [{ index: 1, code: 1 }];
+for (const error of errors) {
+  switch (error.code) {
+    case CreateTransferError.exists: {
+      console.error(`Batch transfer at ${error.index} already exists.`);
+	default:
+	  console.error(`Batch transfer at ${error.index} failed to create: ${CreateTransferError[error.code]}.`);
   }
-  
-  // Two-phase transfer (pending):
-  let flags = 0n
-  flags |= TransferFlags.pending
-
-  // Linked two-phase transfer (pending):
-  let flags = 0n
-  flags |= TransferFlags.linked
-  flags |= TransferFlags.pending
+}
 ```
 
-### Post a Pending Transfer (2-Phase)
+The example above shows that the transfer in index 1 failed with
+error 1. This error here means that `transfer1` and `transfer3` were
+created successfully. But `transfer2` was not created.
+
+To handle errors you can either 1) exactly match error codes returned
+from `client.createTransfers` with enum values in the
+`CreateTransferError` object, or you can 2) look up the error code in
+the `CreateTransferError` object for a human-readable string.
+
+### Batching
+
+TigerBeetle performance is maximized when you batch inserts. The
+client does not do this automatically for you. So, for example, you
+can insert 1 million transfers one at a time like so:
+
+```js
+for (let i = 0; i < 1_000_000; i++) {
+  const errors = client.createTransfers(transfers[i]);
+  // error handling omitted
+}
+```
+
+But the insert rate will be a fraction of potential. Instead, always
+batch what you can. The maximum batch size is set in the TigerBeetle
+server. The default is 8191.
+
+```js
+const BATCH_SIZE = 8191;
+for (let i = 0; i < 1_000_000; i += BATCH_SIZE) {
+  const errors = client.createTransfers(transfers.slice(i, Math.min(transfers.length, BATCH_SIZE)));
+  // error handling omitted
+}
+```
+
+### Transfer Flags
+
+The transfer `flag` value is a bitfield. See details for these flags in
+the [Transfers
+reference](https://docs.tigerbeetle.com/reference/transfers#flags).
+
+To toggle behavior for an transfer, combine enum values stored in the
+CreateTransferFlags object (in TypeScript it is an actual enum) with
+bitwise-or:
+
+* `CreateTransferFlags.linked`
+* `CreateTransferFlags.debits_must_not_exceed_credits`
+* `CreateTransferFlags.credits_must_not_exceed_credits`
+
+For example, to set the following `transfer0` as `linked` to
+`transfer1` and that `transfer0` has the setting
+`debits_must_not_exceed_credits`:
```

Comment:
Can you reword this? "and that ... has the setting" is a little unclear.

Reply:
Yeah that was a struggle :D

Reply:
What about:

> For example, to link `account0` and `account1`, where ` account0` additionally has the `debits_must_not_exceed_credits` constraint:

Reply:
Done